tests/test_freqtradebot.py

```diff
@@ -4308,3 +4309,244 @@ def test_get_valid_price(mocker, default_conf_usdt) -> None:
 
     assert valid_price_at_min_alwd > custom_price_under_min_alwd
     assert valid_price_at_min_alwd < proposed_price
+
+
+def test_position_adjust(mocker, default_conf_usdt, fee) -> None:
```

Comment:
This test is actually not holding what it promises (so i was disappointed at the end :D  )

Naming suggests that it'll test position adjustment - but it actually tests  order filling in case of additional orders - not the adjustment logic itself (neither `process_open_trade_positions()` - nor `check_and_call_adjust_trade_position` are actually called).


what i'd do is the following (maybe in `test_integration.py`)?
- try to mock as little as possible (basically only pricing and "is order filled" - and what signals we need at this point) - stick to dry-run orders (the logic will not change if it's dry or live)
- try to run the "real" logic - basically calling `.process()` - have it enter a trade first - followed by a "DCA order" (you'll have to have the strategy return the correct things).

Reply:
I've given this test a try now in the last commit ... ;)